.circleci/config.yml

```diff
@@ -91,6 +91,18 @@ jobs:
           # fails if jobs not limited; read more in https://github.com/0xSpaceShard/starknet-devnet-rs/issues/378
           command: cargo test --jobs 3 --test '*' --no-fail-fast
 
+  spell-check:
+    docker:
+      - image: cimg/rust:1.75.0
```

Comment:
I'm skeptical about this job being the only one dependent on rust 1.75.0 and everything else using 1.74.0. I see that typos-cli 1.16.26 is the last version using a rust version before 1.75.0 (more specifically, it's dependent on rust 1.70.0), so one option is installing that version, just not sure how much behind with all the features that version is.

The other option is to update the whole toolchain to 1.75.0, but not sure if we want to do that just to support a spelling checker.

The list of versions with rust requirements: https://crates.io/crates/typos-cli/versions

Reply:
Locally I ran this:
```
docker run --rm -it --mount type=bind,source=/home/fabijanc/path/to/starknet-devnet-rs/,target=/home/circleci/project cimg/rust:1.75.0
```

and inside it I ran:
```
$ cargo install typos-cli
$ typos
```

and got this:
```
error: failed to compile `typos-cli v1.20.10`, intermediate artifacts can be found at `/tmp/cargo-install94t5c8`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

Caused by:
  package `typos-vars v0.8.16` cannot be built because it requires rustc 1.75 or newer, while the currently active rustc version is 1.74.0
  Try re-running cargo install with `--locked`
```

(Installing with `--locked` doesn't help)

Reply:
I find that it is because the `rust-toolchain.toml` file will override the version.

Reply:
makes sense, to reduce the impact of your PR, can you try installing the mentioned older version of typos-cli (1.16.26)?

Reply:
Resolved by using the nightly toolchain